zp-relayer/state/PoolState.ts

```diff
@@ -87,7 +84,10 @@ export class PoolState {
     return [outCommit, memo]
   }
 
-  getMerkleRoot(): string {
+  getMerkleRoot(index?: number): string {
+    if (index) {
+      return this.tree.getRootAt(index)
```

Comment:
Will it return an error if an index is greater than `getNextIndex()`?

Reply:
Yes, but probably we should make this error explicit and wrapped in `TxValidationError`. Will update this function.

Reply:
Decided to separate normal `getRoot` and `getRootAt` methods. If `getRootAt` throws an error it will be logged and `null` returned as a result.